pandas/io/tests/test_sql.py

```diff
@@ -665,6 +666,25 @@ def test_date_and_index(self):
         self.assertTrue(issubclass(df.IntDateCol.dtype.type, np.datetime64),
                         "IntDateCol loaded with incorrect type")
 
+    def test_sqlite_time(self):
+        # test support for datetime.time
+        df = DataFrame([time(9, 0, 0), time(9, 1, 30)], columns=["a"])
+        # GH 8341
+        # test if time object are correctly converted into string
+        # in a uniform way between sqlalchemy and the sqlite3 fallback
+        # this uses a sqlite3 adapter
+        if self.flavor == 'sqlite':
+            sql.to_sql(df, "test_time", self.conn, index=False)
+            res = sql.read_sql_query("SELECT * FROM test_time", self.conn)
+            ref = df.applymap(lambda _: _.strftime("%H:%M:%S.%f"))
+            tm.assert_frame_equal(ref, res)
+            # are we sure sqlalchemy was not affected by the adapter
+            engine = TestSQLApi.connect()
+            sql.to_sql(df, "test_time", engine, index=False)
+            res = sql.read_sql_query("SELECT * FROM test_time", engine)
+            ref = df.applymap(lambda _: _.strftime("%H:%M:%S.%f"))
+            tm.assert_frame_equal(ref, res)
```

Comment:
Can you move this test to the respective `test_datetime_time` tests in the flavour test classes? (they already exist, I would update the fallback one with the new functionality, and the sqlalchemy one with the extra test that it keeps working after using the sqlite fallback conn) As I think it is more flavor dependent and not general API (since there is a difference between sqlite fallback and slqalchemy)

I would also do the last test here with `read_sql_table` where the result should be datetime.time values and no coercing to strings should be needed.